packages/gatsby/src/internal-plugins/prod-404/gatsby-node.js

```diff
@@ -1,13 +1,28 @@
-let created404 = false
+const { emitter } = require(`../../redux`)
+const { boundActionCreators } = require(`../../redux/actions`)
+
+const PROD_404_PAGE_PATH = `/404.html`
+
+let page404 = null
 exports.onCreatePage = ({ page, store, actions }) => {
   // Copy /404/ to /404.html as many static site hosts expect
   // site 404 pages to be named this.
   // https://www.gatsbyjs.org/docs/add-404-page/
-  if (!created404 && /^\/?404\/?$/.test(page.path)) {
+  if (!page404 && /^\/?404\/?$/.test(page.path)) {
     actions.createPage({
       ...page,
-      path: `/404.html`,
+      path: PROD_404_PAGE_PATH,
     })
-    created404 = true
+    page404 = page
   }
 }
+
+emitter.on(`DELETE_PAGE`, action => {
+  if (page404 && action.payload.path === page404.path) {
+    boundActionCreators.deletePage({
+      ...page404,
+      path: PROD_404_PAGE_PATH,
+    })
```

Comment:
Can this cause problems because we pass a copy of the page and not the exact same instance. I just don't know if we use strict `===` comparison for pages somewhere or not.

Reply:
We already create "duplicate" internal page (with more metadata) and store this in redux store ( https://github.com/gatsbyjs/gatsby/blob/08447bd4d13b1f2246daca158d828c2b04482067/packages/gatsby/src/redux/actions/public.js#L381-L423 )

So even if we stored exact argument of `createPage`, it wouldn't be the same instance as one we keep in state